diff --git a/python/test_vending_machine.py b/python/test_vending_machine.py
index 2a5d600..36e5814 100644
--- a/python/test_vending_machine.py
+++ b/python/test_vending_machine.py
@@ -33,3 +33,12 @@ def test_accept_coins(machine: VendingMachine, printer: VendingMachinePrinter, c
     assert machine.balance == 5
     assert machine.coins == [5]
     assert machine.display == "5"
+
+
+def test_return_coins(machine: VendingMachine, printer: VendingMachinePrinter):
+    # TODO: use the printer and approvaltests.verify instead of assertions
+
+    machine.return_coins()
+
+    assert machine.balance == 0
+
diff --git a/python/vending_machine.py b/python/vending_machine.py
index b2337e6..505d663 100644
--- a/python/vending_machine.py
+++ b/python/vending_machine.py
@@ -21,3 +21,9 @@ class VendingMachine:
             self._display_balance()
         else:
             self.returns.append(value)
+
+    def return_coins(self):
+        self.balance = 0
+        self.returns = self.coins
+        self.coins = []
+        self._display_balance()
